### PR TITLE
224 correct vignette visualization deployment

### PR DESCRIPTION
**Pull request by PietrH**
The vignette: `visualize-deployment-features.Rmd` has gotten out of date, zero count observations are displayed by default as black multiplication symbols `×`, not red circle markers. 

After discussion with Damiano we've decided to update the vignette, not the function `map_dep()`. 

The function used to display red circle markers for zero count observations, but was updated to use an icon instead. This vignette was overlooked. 

image

---

Damiano noticed in this PR that an older issue is actually unsolved: https://github.com/inbo/camtraptor/issues/6#issuecomment-1644065206 : 

- [ ] `map_dep()` can differentiate between NA and 0
- [ ] `map_dep()` visualizes NA and 0 differently 
`map_dep()` offers users control over NA and 0 visualization:
- [ ] `na_values_show`
- [ ] `na_values_icon_url`
- [ ] `na_values_icon_size`
Assertions were added for these last arguments.

### CODE REVIEW

**Comment by PietrH** [comment]
@damianooldoni Could you have a look at the rest of the vignette? Especially the mentions of the colours red or grey.

**Review by damianooldoni — APPROVED** [review]
Thansk @PietrH to ask me to review the vignette. It was time to update it. And the difference between NAs and 0 while visualizing the number of species was gone, apparently. I think I fixed it. 
I would like your review. But I cannot point you as reviewer officially as you are the owner of this PR 🤣

File: tests/testthat/test-map_dep.R [hunk 1]
```diff
@@ -72,6 +81,36 @@ test_that("map_dep() can toggle showing deployments with zero values", {
...
+  expect_error(map_dep(mica, feature = "n_obs",
+                       na_values_show = c(TRUE, TRUE)),
+               regexp = "na_values_show must have length 1.")
+  # expect an error when the toggle is not TRUE or FALSE
+  expect_error(map_dep(mica, feature = "n_obs",
+                       na_values_show = "dax"),
+               regexp = "na_values_show must be a logical: TRUE or FALSE.")
+  expect_error(map_dep(mica, feature = "n_obs",
+                       na_values_show = NA),
+               regexp = "na_values_show must be a logical: TRUE or FALSE.")
```
Comment (PietrH) [inline tests/testthat/test-map_dep.R:97]: minor suggestion: set `fixed = TRUE` when using literal matching

**Review by PietrH — COMMENTED** [review]
@damianooldoni, I've suggested some minor changes. 

Mostly about test coverage, if you want we can split that off to a different pull request. In that last case I can contribute. 

Do you think we should test the output of `map_dep()` for certain properties, such as what leaflet calls were used and so forth? Such a test might notify us of problems such as not plotting things we expect to be plotted, but is less strict than a snapshot.

File: R/map_dep.R [hunk 2]
```diff
@@ -380,6 +396,34 @@ map_dep <- function(package = NULL,
...
+    is.logical(zero_values_show) & !is.na(zero_values_show),
+    msg = "zero_values_show must be a logical: TRUE or FALSE."
+  )
+  assertthat::assert_that(
+    zero_values_show %in% c(TRUE, FALSE),
+    msg = "zero_values_show must be a logical: TRUE or FALSE."
+  )
+  
+  # check na_values_show is a toggle (TRUE or FALSE)
+  assertthat::assert_that(
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/map_dep.R:406]: Thanks to notice this! Nice series of functions. I will use it here as well.

File: tests/testthat/test-map_dep.R [hunk 3]
```diff
@@ -54,8 +54,17 @@ test_that("map_dep() can handle combinations of arguments", {
 })
 
 test_that("map_dep() can toggle showing deployments with zero values", {
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply tests/testthat/test-map_dep.R:56]: mmhh, I will check better the entire if-else statement as it makes not completely sense.

[the remark this answers is not in the document]
Comment (damianooldoni) [reply tests/testthat/test-map_dep.R:56]: Done in [8363f6e](https://github.com/inbo/camtraptor/pull/229/commits/8363f6e7acbe8a068aae059accd821f59523bbb1)

File: tests/testthat/test-map_dep.R [hunk 4]
```diff
@@ -72,6 +81,36 @@ test_that("map_dep() can toggle showing deployments with zero values", {
   expect_no_message(
     map_dep(mica,feature = "n_species", zero_values_show = TRUE)
   )
+})
+
+test_that("map_dep() can toggle showing deployments with NA values", {
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply tests/testthat/test-map_dep.R:93]: Your idea makes me think `get_n_species()` is poorly tested. So, I will add tests about 0 observations and NA obs if `test-get_n_species.R`

[the remark this answers is not in the document]
Comment (damianooldoni) [reply tests/testthat/test-map_dep.R:93]: Done in [46158b2](https://github.com/inbo/camtraptor/pull/229/commits/46158b28a230bf80ff3170c98377a2b1700f6ed5) and [f2220a3](https://github.com/inbo/camtraptor/pull/229/commits/f2220a375a59f7fccfc9521a97ba0fea4693b004). I would add tests for leaflet maybe later in another PR. But I would do it while working on adapting `map_dep()` (see https://github.com/inbo/camtraptor/issues/91: we didn't speak about it during the sprint, but I think would be important to tackle as well)

**Pull request merged**